### PR TITLE
Phase 3 funding | Aperio | GUI-editable regions

### PR DESCRIPTION
**Pull request by keflavich**
Proposal to finish the work on making regions GUI-editable.

### CODE REVIEW

**Comment by keflavich** [comment]
@dhomeier I'd like your direct input on this - could you help assess what an appropriate level of effort is?

@Cadair your input might also be helpful from the management perspective.

**Comment by dhomeier** [comment]
> @dhomeier I'd like your direct input on this - could you help assess what an appropriate level of effort is?

I think so far we have spent some 150 hours mainly on the `regions` side, which is now primarily waiting on a finalised `matplotlib` API. But all the more important to have the hours to adapt the `regions` PRs to that once finished.
@dstansby may have a better overview how much work is yet to do in `matplotlib.widgets`.

Scope of the documentation part is probably fairly open.

**Comment by dhomeier** [comment]
> The blocker when I did this work previously was disagreement on the MPL side on how rectangles should be rotated, but this is now resolved: [matplotlib/matplotlib#21945 (comment)](https://github.com/matplotlib/matplotlib/pull/21945#issuecomment-1165410652)

It was not clear to me from that comment if changing the implementation to "_stay a rectangle in display coordinates_" was mostly or largely done, or needed substantial coding yet. If the former, the work should hopefully be on a comparable level to what is needed in `regions` itself.

**Comment by jdswinbank** [comment]
Because the estimated total of all the proposals is a bit higher than the amount available for allocation, we suggest that you (@keflavich ) provide a budget range - i.e., a "minimum for this project to be viable" and "maximum you could use effectively for this project".

**Comment by dhomeier** [comment]
Depending on the level of familiarity with the present `matplotlib` implementation I suppose the 50 h minimum should probably allow adapting that and the existing `regions` PRs, without any further documentation.

**Comment by pllim** [comment]
FWIW, Jdaviz has a plugin to edit region shapes already (caveat: limited shapes support, needs to go through glue).

**Comment by pllim** [comment]
Ginga also has shape edit capability (caveat: starts out as Ginga shape object but it has translator for regions).

https://ginga.readthedocs.io/en/latest/manual/plugins_local/drawing.html

https://ginga.readthedocs.io/en/latest/dev_manual/canvas.html#support-for-astropy-regions

File: finance/proposal-calls/cycle3/editable-regions.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+### Title
+Complete effort on gui-editable reginos
```
Comment (astrofrog) [inline finance/proposal-calls/cycle3/editable-regions.md:2]:
```suggestion
Complete effort on gui-editable regions
```

File: finance/proposal-calls/cycle3/editable-regions.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+### Project Team
+Adam Ginsburg
+Derek Homeier
+Aperio
```
Comment (astrofrog) [inline finance/proposal-calls/cycle3/editable-regions.md:7]:
```suggestion
Aperio Software
```

File: finance/proposal-calls/cycle3/editable-regions.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+that need to be wrapped up.  Then, significant documentation is needed to show users how to use these cool tools.
+
+### Project / Work
+The main work is to complete the above PRs and write new documentation
+
+### Approximate Budget
+Currency: US $20,000 estimated.
+
+All will be used to cover effort by Aperio team members.
```
Comment (astrofrog) [inline finance/proposal-calls/cycle3/editable-regions.md:31]:
```suggestion
All will be used to cover effort by Aperio Software team members.
```

**Comment by kelle** [comment]
Please react to this comment to vote on this proposal (👍, 👎, or no reaction for +0).

**Comment by jdswinbank** [comment]
Congratulations!  The community has expressed support for this FR, and the Finance Committee has determined it can be funded.  This FR has been approved for $20,000, to be funded from the NASA grant.

@kelle is the Finance Committee member who will be supporting this project.  They will update this PR with the amount and then merge it, as well as being in contact with you to set up any necessary agreements and relevant Open Collective financial arrangements. They will also open a new issue where progress on this project can be discussed.

**Pull request merged**